### PR TITLE
Big form

### PR DESCRIPTION
**Pull request by oryrosin**

### CODE REVIEW

File: src/Screens.jsx [hunk 1]
```diff
@@ -49,7 +49,19 @@ const Screens = () => {
         <Route path="/temp" element={<MobileScreen />} />
```
Comment (alonisser) [inline src/Screens.jsx:49]: what does this "temp" screen do?

File: src/apiHooks/createRegistrationDetails.jsx [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 
 export default async function createProfile({
+  id,
   date_of_birth,
   self_identified_gender,
   academic_affiliation,
   country_of_residence,
   academic_stage,
-  has_ASSC_membership = true,
+  has_opted_for_contrast_updates = true,
```
Comment (alonisser) [inline src/apiHooks/createRegistrationDetails.jsx:10]: why instead? assc , shouldn't it be "besides"?

File: src/pages/Register/SecondaryRegister.jsx [hunk 3]
```diff
@@ -9,20 +9,40 @@ import { useSnapshot } from "valtio";
...
+import getUser from "../../apiHooks/getUser";
+import { format } from "date-fns/esm";
 
 export default function SecondaryRegister() {
   const [errorMsg, setErrorMsg] = useState("");
   const [dateError, setDateError] = useState(false);
+  const [isNew, setIsNew] = useState(true);
+  const [searchParams] = useSearchParams();
+  const isNewUser = searchParams.get("new");
+  console.log(isNewUser);
```
Comment (alonisser) [inline src/pages/Register/SecondaryRegister.jsx:24]: why console log

File: src/pages/Register/SecondaryRegister.jsx [hunk 4]
```diff
@@ -58,16 +84,15 @@ export default function SecondaryRegister() {
...
         country_of_residence: values.country,
         academic_stage: values.academicStage,
         check: values.check,
       });
-      if (result.status === 201) {
-        navigate("/");
-      }
+      isNewUser ? navigate("/") : navigate(-1);
     } catch (e) {
       e.response?.status === 400 && setErrorMsg(e.response?.data);
```
Comment (alonisser) [inline src/pages/Register/SecondaryRegister.jsx:97]: and if not? toaster?

File: src/pages/Register/SecondaryRegister.jsx [hunk 5]
```diff
@@ -94,119 +119,125 @@ export default function SecondaryRegister() {
...
+                        <Field
+                          as="select"
+                          id="day"
+                          name="day"
+                          className={fieldClass + "w-20"}>
+                          <option value=""> Day</option>
+                          {generateSelectOptions(1, 31)}
+                        </Field>
+
+                        <Field
```
Comment (alonisser) [inline src/pages/Register/SecondaryRegister.jsx:146]: this is a birth date selector? if so, let the initial value be 1990, they can edit it after that, but it's a sane place to start, especially if there is scrolling

File: src/Screens.jsx (hunk 1 above)
Comment (oryrosin) [reply src/Screens.jsx:49]: allows the developer try things

**Review by alonisser — DISMISSED** [review]

**Review by alonisser — APPROVED** [review]

**Review by alonisser — APPROVED** [review]

**Pull request merged**